### PR TITLE
v1.7.0 - updated BLAST to 2.13.0+ and fixed long rec IDs

### PR DESCRIPTION
**Pull request by dcchivian**

### CODE REVIEW

**Comment by dcchivian** [comment]
@Tianhao-Gu here are the deltas categorized by what aspect of the update they correspond to. I didn't include the LGTM-recommended style updates. Also didn't include redundant deltas, just the first instance. I sometimes had difficulty getting the line(s) to open from the URL, so let me know if you would rather have the filename and line numbers.

Update to BLAST v2.13.0
-----------------------
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R11-R16
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R20-R23
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-8ae074e0e61c5093be13bf4f1cde9091390969d444fa2310f873fb631b2a8d29R4
 

Feature ID length fix
---------------------
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-6fba2e3214cc8eb834af7078cfd9f460ea6d143908f352a1c43abde402a4834aR327
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-6fba2e3214cc8eb834af7078cfd9f460ea6d143908f352a1c43abde402a4834aR559
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-6fba2e3214cc8eb834af7078cfd9f460ea6d143908f352a1c43abde402a4834aR602
* 
https://github.com/kbaseapps/kb_blast/pull/66/files#diff-6fba2e3214cc8eb834af7078cfd9f460ea6d143908f352a1c43abde402a4834aR648
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-6fba2e3214cc8eb834af7078cfd9f460ea6d143908f352a1c43abde402a4834aR746
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-6fba2e3214cc8eb834af7078cfd9f460ea6d143908f352a1c43abde402a4834aR746


Alternate genetic code and underscores in translation
-----------------------------------------------------
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-ab62d6231f2e2ea3656781d06ef196e1179998e98c033fc87131dc167b174e32R23-R24
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-ab62d6231f2e2ea3656781d06ef196e1179998e98c033fc87131dc167b174e32R42
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-6fba2e3214cc8eb834af7078cfd9f460ea6d143908f352a1c43abde402a4834aR249-R253
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-6fba2e3214cc8eb834af7078cfd9f460ea6d143908f352a1c43abde402a4834aR328
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-6fba2e3214cc8eb834af7078cfd9f460ea6d143908f352a1c43abde402a4834aR372-R376
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-3f510e528cad0d6c0acc483af0d862bee160a61d07eabc17a57e50417b93c4c3R305
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-45f8994fa6dfe8757ea59f0dc53d4ae8e0f6b5bc687e1c37adb6521452f08a24R85-R89
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-45f8994fa6dfe8757ea59f0dc53d4ae8e0f6b5bc687e1c37adb6521452f08a24R127-R128
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-5dadc85367a53f3ef82e1cbbff50137e1d8c05e57097ac358ba2b7d783256ffaR163-R186
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-5dadc85367a53f3ef82e1cbbff50137e1d8c05e57097ac358ba2b7d783256ffaR319-R322


Species Tree as target
----------------------
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-6fba2e3214cc8eb834af7078cfd9f460ea6d143908f352a1c43abde402a4834aR404
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-6fba2e3214cc8eb834af7078cfd9f460ea6d143908f352a1c43abde402a4834aR666-R715
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-6fba2e3214cc8eb834af7078cfd9f460ea6d143908f352a1c43abde402a4834aR1407
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-6fba2e3214cc8eb834af7078cfd9f460ea6d143908f352a1c43abde402a4834aR1626
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-6fba2e3214cc8eb834af7078cfd9f460ea6d143908f352a1c43abde402a4834aR1648
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-3f510e528cad0d6c0acc483af0d862bee160a61d07eabc17a57e50417b93c4c3R203-R269
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-3f510e528cad0d6c0acc483af0d862bee160a61d07eabc17a57e50417b93c4c3R572-R629
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-8ae074e0e61c5093be13bf4f1cde9091390969d444fa2310f873fb631b2a8d29R50
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-8ae074e0e61c5093be13bf4f1cde9091390969d444fa2310f873fb631b2a8d29R95
* https://github.com/kbaseapps/kb_blast/pull/66/files#diff-8ae074e0e61c5093be13bf4f1cde9091390969d444fa2310f873fb631b2a8d29R108

File: lib/kb_blast/Utils/BlastUtil.py [hunk 1]
```diff
@@ -656,6 +663,56 @@ def write_target_obj_to_file (self, params, input_many_ref, seq_type):
...
+                'case':                'upper',
+                'linewrap':            50,
+                'id_len_limit':        49,
+                'write_off_code_prot_seq': params['write_off_code_prot_seq'],
+                'merge_fasta_files':   'TRUE'
+                }
+
+            #self.log(console,"callbackURL='"+self.callbackURL+"'")  # DEBUG
+            #SERVICE_VER = 'release'
+            SERVICE_VER = 'beta'  # DEBUG
```
Comment (Tianhao-Gu) [inline lib/kb_blast/Utils/BlastUtil.py]: should we make `SERVICE_VER` a global var? Otherwise, different versions of DOTFU might be used.

**Review by Tianhao-Gu — APPROVED** [review]
👍  LGTM.

File: lib/kb_blast/Utils/BlastUtil.py (hunk 1 above)
Comment (dcchivian) [reply lib/kb_blast/Utils/BlastUtil.py]: @Tianhao-Gu I addressed your concerns in commit 11d5966da6f9a24222ad58a5abc4c0655f9a5d7a.  Please re-review.

**Review by Tianhao-Gu — APPROVED** [review]
LGTM

**Comment by dcchivian** [comment]
Github actions unit tests passing against released KBaseDataObjectToFileUtils.  PR reviewed and passed.  Merging.

**Pull request merged**